source/common/router/router.cc

```diff
@@ -687,14 +687,18 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   // a backoff timer.
   ASSERT(upstream_requests_.size() <= 1);
 
-  bool buffering = (retry_state_ && retry_state_->enabled()) || !active_shadow_policies_.empty();
+  bool buffering =
+      (retry_state_ && retry_state_->enabled()) || !active_shadow_policies_.empty() ||
+      (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.internal_redirects_with_body") &&
```

Comment:
Is the fact that we check this flag in multiple places going to cause consistency issues? i.e. the runtime flag is flipped during a slow request so we only buffer parts of the request? Should we be snapping this runtime evaluation to the filter so that it will be consistent for the duration of a stream?

Reply:
hmm, could be with a very slow request that the first call (`decodeData`) sees a different value than the second call (`setupRedirect` called by `onUpstreamHeaders`) which would either mean attempting the redirect without having the body buffered if flipped off->on or buffering the body then seeing it is disabled when flipped off->on.

I've added a boolean to the filter to store the value and set it in `decodeHeaders`.